src/conversions/rgb_to_cmyk.rs

```diff
@@ -0,0 +1,57 @@
+// Simple RGB to CMYK conversion. Returns percentages of CMYK paint.
+// https://www.programmingalgorithms.com/algorithm/rgb-to-cmyk/
+
+// Note: this is a very popular algorithm that converts colors linearly and gives
+// only approximate results. Actual preparation for printing requires advanced color
+// conversion considering the color profiles and parameters of the target device.
+
+pub fn rgb_to_cmyk(r_input: u8, g_input: u8, b_input: u8) -> (i32, i32, i32, i32) {
+    // Convert RGB values to the range 0..1
+    let r = r_input as f32 / 255.0;
+    let g = g_input as f32 / 255.0;
+    let b = b_input as f32 / 255.0;
```

Comment:
Please define a separate function doing such _mapping_ to `[0.0, 1.0]`. It should also perform some _input check_.